client/src/menu/game/GameScreen.tsx

```diff
@@ -164,6 +194,15 @@ export default function GameScreen(): ReactElement {
         ["addChatMessages"]
     )!;
 
+    useEffect(() => {
+        const onBeforeUnload = (e: BeforeUnloadEvent) => {
+            if (!DEV_ENV) e.preventDefault()

```

Comment:
ok i see
I looked this up and couldnt test it because i couldnt even get in a lobby
I see the idea
Its weird that this is how you call that, can we put our own custom text?
I would like it so that if you confirm that you want to leave it sends a message to the server saying that you left
This way it kills your player in game

Reply:
I don't think so, and I don't think there's a way to make code run when they confirm.
Maybe if you do a window.confirm? But we can do that later